exec/java-exec/src/test/java/org/apache/drill/test/rowSet/RowSetBatch.java

```diff
@@ -29,13 +30,28 @@
 import org.apache.drill.exec.record.selection.SelectionVector2;
 import org.apache.drill.exec.record.selection.SelectionVector4;
 
+import java.util.ArrayList;
 import java.util.Iterator;
+import java.util.List;
 
-public class RowSetBatch implements RecordBatch {
-  private final RowSet rowSet;
+/**
+ * A mock operator that returns the provided {@link RowSet}s as batches. Currently it's assumed that all the {@link RowSet}s have the same schema.
+ */
+public class RowSetBatch implements CloseableRecordBatch {
```

Comment:
RowSetBatch predates MockRecordBatch :/ . But I'll see if I can make MockRecordBatch play nice with RowSets. Since MockRecordBatch is used for many many tests, I will make it work with RowSets while keeping some deprecated constructors for VectorContainers so that I don't have to update every test.

Reply:
If you want you can leave it as is for now. I can take a look on how to refactor MockRecordBatch class along with other tests which uses it, which I have to do anyways. I was also thinking of making it use RowSets instead of container. 
The reason for creating separate class was to use it specifically for testing different IterOutcomes in context of Lateral&Unnest only whereas RowSetBatch looked more of a very simple implementation to just hold one container. Was not sure if MockRecordBatch will play well with other operators in general.

Reply:
@sohami It's okay, I've already made the move to MockRecordBatch for this PR. Just doing a final test runs, and debugging straggling issues.